src/platform/packages/shared/shared-ux/error_boundary/src/ui/section_error_boundary.tsx

```diff
@@ -0,0 +1,102 @@
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React from 'react';
+
+import type { KibanaErrorBoundaryServices } from '../../types';
+import { useErrorBoundary } from '../services/error_boundary_services';
+import { SectionFatalPrompt, SectionRecoverablePrompt } from './message_components';
+
+interface SectionErrorBoundaryProps {
+  sectionName: string;
+}
+
+/**
+ * `KibanaSectionErrorBoundary` is designed to capture errors at a granular level.
+ *
+ * In general, it's best to use `KibanaErrorBoundary` and block the whole page.
+ * Users will see an error state on the page and think that there are instabilities in the system.
+ * They will be / should be wary about making any changes in a UI showing an error, since it risks
+ * further instability.
+ *
+ * If it is acceptable to assume the risk of allowing users to interact with a UI that
+ * has an error state, then using `KibanaSectionErrorBoundary` may be an acceptable alternative,
+ * but this must be judged on a case-by-case basis.
+ */
+export const KibanaSectionErrorBoundary = (
+  props: React.PropsWithChildren<SectionErrorBoundaryProps>
+) => {
+  const services = useErrorBoundary();
+  return <SectionErrorBoundaryInternal {...props} services={services} />;
+};
+
+interface ErrorBoundaryState {
+  error: null | Error;
+  errorInfo: null | Partial<React.ErrorInfo>;
+  componentName: null | string;
+  isFatal: null | boolean;
+}
+
+const DEFAULT_STATE = {
+  error: null,
+  errorInfo: null,
+  componentName: null,
+  isFatal: null,
+} as const;
+
+interface ServiceContext {
+  services: KibanaErrorBoundaryServices;
+}
+
+class SectionErrorBoundaryInternal extends React.Component<
+  React.PropsWithChildren<SectionErrorBoundaryProps> & ServiceContext,
+  ErrorBoundaryState
+> {
+  constructor(props: SectionErrorBoundaryProps & ServiceContext) {
+    super(props);
+
+    this.state = DEFAULT_STATE;
+  }
+
+  componentDidCatch(error: Error, errorInfo: React.ErrorInfo): void {
+    console.error('Error caught by Kibana React Section Error Boundary'); // eslint-disable-line no-console
+    console.error(error); // eslint-disable-line no-console
+
+    const { name, isFatal } = this.props.services.errorService.registerError(error, errorInfo);
+    this.setState({ error, errorInfo, componentName: name, isFatal });
+  }
+
+  render() {
+    if (!this.state.error) {
+      return this.props.children;
+    }
+
+    const { error, errorInfo, componentName, isFatal } = this.state;
+
+    if (isFatal) {
+      return (
+        <SectionFatalPrompt
+          sectionName={this.props.sectionName}
+          error={error}
+          errorInfo={errorInfo}
+          name={componentName}
+        />
+      );
+    }
+
+    return (
+      <SectionRecoverablePrompt
+        sectionName={this.props.sectionName}
+        onRecoverAttempt={this.resetState}
```

Comment:
I've mentioned in my previous comment, this prop should use the original name of `onClickRefresh` and the callback used should be `this.props.services.onClickRefresh`

```suggestion
        onClickRefresh={this.props.services.onClickRefresh}
```